src/main/java/me/arminb/sara/dao/UserDAOImpl.java

```diff
@@ -30,47 +33,40 @@
     @Autowired
     private MongoDatabase database;
 
-    @Override
-    public List<User> findAll(Integer pageNumber, Integer pageCount) throws DataAccessException {
-        if (pageNumber == null){
-            pageNumber = PAGE_NUMBER;
-        }
-        if (pageCount == null){
-            pageCount = PAGE_COUNT;
-        }
-        List<User> list = new ArrayList();
+    private User mapUser(Document userDoc){
+        User user = new User();
+        DateFormat format = new SimpleDateFormat("EEE MMM d HH:mm:ss z yyyy");
+        Date createDate = null;
+        Date modifiedDate = null;
         try {
-            MongoCollection<Document> collection = database.getCollection("users");
-            MongoCursor<Document> iterator = collection.find().skip(pageCount * (pageNumber - 1)).limit(pageCount).iterator();
-            while (iterator.hasNext()) {
-                String jsonInString = iterator.next().toJson();
-                Gson g = new GsonBuilder().create();
-                User user = g.fromJson(jsonInString, User.class);
-                list.add(user);
+            if (userDoc.get("created_date") != null){
+            createDate = format.parse(userDoc.get("created_date").toString());
             }
-            if (list.size() != 0) {
-                return list;
-            }
-            else {
-                return null;
+            if (userDoc.get("modified_date") != null) {
+                modifiedDate = format.parse(userDoc.get("modified_date").toString());
             }
-        } catch (MongoException e) {
-            logger.warn("Failed to fetch users from the database", e);
-            throw new DataAccessException();
+        } catch (ParseException e) {
+            e.printStackTrace();
```

Comment:
never use `e.printStackTrace();`. If you can ignore the exception just log it as WARN and move on. Otherwise do something about it. Here modifiedAt should be set to null. However, because it's been set to null at the beginning you can just do a warn log and move on.